packages/langium/src/parser/cst-node-builder.ts

```diff
@@ -55,6 +54,33 @@ export class CstNodeBuilder {
         }
     }
 
+    addHiddenNode(token: IToken): void {
+        const leafNode = new LeafCstNodeImpl(token.startOffset, token.image.length, tokenToRange(token), token.tokenType, true);
+        leafNode.root = this.rootNode;
+        let current: CompositeCstNode = this.current;
+        let added = false;
+        // If we are within a composite node, we add the hidden node to the content
+        if (current.content.length > 0) {
+            current.content.push(leafNode);
+            return;
+        }
+        while (current.container) {
+            const index = current.container.content.indexOf(current);
```

Comment:
Question: I thought I saw a performance improvement, but then I saw that `$containerIndex` is only defined on `AstNode`. Is there a reason why the `CstNode` does not have a `$containerIndex` property? This would be a difference from `O(N)` to `O(1)` here, if the given `N` is the number of children.